# Loading the dataset and getting the insite from it.

In [121]:
# Load the required lib
import pandas as pd
import numpy as np

In [122]:
# Loading the dataset
df1 = pd.read_csv('D:\Data Science\Visualization Project\WorldCup-EDA\Dataset\WorldCupMatches.csv')
df1.head()

Year              Datetime    Stage         Stadium         City  \
0  1930.0  13 Jul 1930 - 15:00   Group 1         Pocitos  Montevideo    
1  1930.0  13 Jul 1930 - 15:00   Group 4  Parque Central  Montevideo    
2  1930.0  14 Jul 1930 - 12:45   Group 2  Parque Central  Montevideo    
3  1930.0  14 Jul 1930 - 14:50   Group 3         Pocitos  Montevideo    
4  1930.0  15 Jul 1930 - 16:00   Group 1  Parque Central  Montevideo    

  Home Team Name  Home Team Goals  Away Team Goals Away Team Name  \
0         France              4.0              1.0         Mexico   
1            USA              3.0              0.0        Belgium   
2     Yugoslavia              2.0              1.0         Brazil   
3        Romania              3.0              1.0           Peru   
4      Argentina              1.0              0.0         France   

  Win conditions  Attendance  Half-time Home Goals  Half-time Away Goals  \
0                     4444.0                   3.0                   0.0   
1                    18346.0                   2.0                   0.0   
2                    24059.0                   2.0                   0.0   
3                     2549.0                   1.0                   0.0   
4                    23409.0                   0.0                   0.0   

                  Referee               Assistant 1  \
0  LOMBARDI Domingo (URU)     CRISTOPHE Henry (BEL)   
1       MACIAS Jose (ARG)  MATEUCCI Francisco (URU)   
2     TEJADA Anibal (URU)   VALLARINO Ricardo (URU)   
3   WARNKEN Alberto (CHI)       LANGENUS Jean (BEL)   
4     REGO Gilberto (BRA)      SAUCEDO Ulises (BOL)   

                  Assistant 2  RoundID  MatchID Home Team Initials  \
0         REGO Gilberto (BRA)    201.0   1096.0                FRA   
1       WARNKEN Alberto (CHI)    201.0   1090.0                USA   
2         BALWAY Thomas (FRA)    201.0   1093.0                YUG   
3    MATEUCCI Francisco (URU)    201.0   1098.0                ROU   
4  RADULESCU Constantin (ROU)    201.0   1085.0                ARG   

  Away Team Initials  
0                MEX  
1                BEL  
2                BRA  
3                PER  
4                FRA

The FIFA World Cup is a global football competition contested by the various football-playing nations of the world. It is contested every four years and is the most prestigious and important trophy in the sport of football.

The World Cups dataset show all information about all the World Cups in the history, while the World Cup Matches dataset shows all the results from the matches contested as part of the cups.

1. year : The year in which the match was played.
2. Datetime : The date at which match was played along with 24 hrs date format.
3. Stage : The stage at which the match was played.
4. Stadium : Satdium name where the match was held.
5. city : The city name where the match was played.
6. Home team name : Home team country name.
7. Home team gaol : Total goals scored by the home team by the end of the match.
8. Away Team Goals : Total goals scored by the away team by the end of the match.
9. Away Team Name : Away team country name.
10. Win conditions : Special win condition (if any)
11. Attendance : Total crowd present at the satdium
12. Half-time Home Goals : Goals scored by the home team until half time
13. Half-time Away Goals : Goals scored by the away team until half time
14. Referee : Name of the first refree
15. Assistant 1 : Name of the first assistant referee (linesman)
16. Assistant 2 : Name of the second assistant referee (linesman)
17. RoundID : Unique ID of the Round
18. MatchID : Unique ID of the match
19. Home Team Initials : Home team country's three letter initials
20. Away Team Initials : Away team country's three letter initials

In [123]:
df2 = pd.read_csv('D:\Data Science\Visualization Project\WorldCup-EDA\Dataset\WorldCupPlayers.csv')
df2.head()

RoundID  MatchID Team Initials           Coach Name Line-up  Shirt Number  \
0      201     1096           FRA  CAUDRON Raoul (FRA)       S             0   
1      201     1096           MEX     LUQUE Juan (MEX)       S             0   
2      201     1096           FRA  CAUDRON Raoul (FRA)       S             0   
3      201     1096           MEX     LUQUE Juan (MEX)       S             0   
4      201     1096           FRA  CAUDRON Raoul (FRA)       S             0   

        Player Name Position Event  
0       Alex THEPOT       GK   NaN  
1   Oscar BONFIGLIO       GK   NaN  
2  Marcel LANGILLER      NaN  G40'  
3      Juan CARRENO      NaN  G70'  
4   Ernest LIBERATI      NaN   NaN

1. RoundID : Unique ID of the round
2. MatchID : Unique ID of the match
3. Team Initials : Player's team initials
4. Coach Name : Name and country of the team coach
5. Line-up : S=Line-up, N=Substitute
6. Shirt Number : Shirt number if available 
7. Player Name : Name of the player
8. Position: C=Captain, GK=Goalkeeper 
9. Event : G=Goal, OG=Own Goal, Y=Yellow Card, R=Red Card, SY = Red Card by second yellow, P=Penalty, MP=Missed Penalty, I = Substitution In, O=Substitute Out 


In [124]:
df3 = pd.read_csv('D:\Data Science\Visualization Project\WorldCup-EDA\Dataset\WorldCups.csv')
df3.head()

Year      Country      Winner      Runners-Up    Third      Fourth  \
0  1930      Uruguay     Uruguay       Argentina      USA  Yugoslavia   
1  1934        Italy       Italy  Czechoslovakia  Germany     Austria   
2  1938       France       Italy         Hungary   Brazil      Sweden   
3  1950       Brazil     Uruguay          Brazil   Sweden       Spain   
4  1954  Switzerland  Germany FR         Hungary  Austria     Uruguay   

   GoalsScored  QualifiedTeams  MatchesPlayed Attendance  
0           70              13             18    590.549  
1           70              16             17    363.000  
2           84              15             18    375.700  
3           88              13             22  1.045.246  
4          140              16             26    768.607

1. Year : Year of the worldcup
2. Country : Country of the worldcup
3. Winner : Team who won the worldcup
4. Runners-Up : Team who was the second place
5. Third : Team who was the third place
6. Fourth : Team who was the fourth place
7. GoalsScored : Total goals scored in the worldcup
8. QualifiedTeams : Total participating teams
9. MatchesPlayed : Total matches played in the cup
10. Attendance : Total attendance of the worldcup


## Feature engg.

In [125]:
df1.head(5)

Year              Datetime    Stage         Stadium         City  \
0  1930.0  13 Jul 1930 - 15:00   Group 1         Pocitos  Montevideo    
1  1930.0  13 Jul 1930 - 15:00   Group 4  Parque Central  Montevideo    
2  1930.0  14 Jul 1930 - 12:45   Group 2  Parque Central  Montevideo    
3  1930.0  14 Jul 1930 - 14:50   Group 3         Pocitos  Montevideo    
4  1930.0  15 Jul 1930 - 16:00   Group 1  Parque Central  Montevideo    

  Home Team Name  Home Team Goals  Away Team Goals Away Team Name  \
0         France              4.0              1.0         Mexico   
1            USA              3.0              0.0        Belgium   
2     Yugoslavia              2.0              1.0         Brazil   
3        Romania              3.0              1.0           Peru   
4      Argentina              1.0              0.0         France   

  Win conditions  Attendance  Half-time Home Goals  Half-time Away Goals  \
0                     4444.0                   3.0                   0.0   
1                    18346.0                   2.0                   0.0   
2                    24059.0                   2.0                   0.0   
3                     2549.0                   1.0                   0.0   
4                    23409.0                   0.0                   0.0   

                  Referee               Assistant 1  \
0  LOMBARDI Domingo (URU)     CRISTOPHE Henry (BEL)   
1       MACIAS Jose (ARG)  MATEUCCI Francisco (URU)   
2     TEJADA Anibal (URU)   VALLARINO Ricardo (URU)   
3   WARNKEN Alberto (CHI)       LANGENUS Jean (BEL)   
4     REGO Gilberto (BRA)      SAUCEDO Ulises (BOL)   

                  Assistant 2  RoundID  MatchID Home Team Initials  \
0         REGO Gilberto (BRA)    201.0   1096.0                FRA   
1       WARNKEN Alberto (CHI)    201.0   1090.0                USA   
2         BALWAY Thomas (FRA)    201.0   1093.0                YUG   
3    MATEUCCI Francisco (URU)    201.0   1098.0                ROU   
4  RADULESCU Constantin (ROU)    201.0   1085.0                ARG   

  Away Team Initials  
0                MEX  
1                BEL  
2                BRA  
3                PER  
4                FRA

Loading the top 5 records 

In [126]:
df1.shape

(4572, 20)

From the above I can see that there are 4572 rows and 20 columns.

In [127]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4572 entries, 0 to 4571
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Year                  852 non-null    float64
 1   Datetime              852 non-null    object 
 2   Stage                 852 non-null    object 
 3   Stadium               852 non-null    object 
 4   City                  852 non-null    object 
 5   Home Team Name        852 non-null    object 
 6   Home Team Goals       852 non-null    float64
 7   Away Team Goals       852 non-null    float64
 8   Away Team Name        852 non-null    object 
 9   Win conditions        852 non-null    object 
 10  Attendance            850 non-null    float64
 11  Half-time Home Goals  852 non-null    float64
 12  Half-time Away Goals  852 non-null    float64
 13  Referee               852 non-null    object 
 14  Assistant 1           852 non-null    object 
 15  Assistant 2          

From the above i can see that there are names of the column and there not null data and its data type

In [128]:
df1.describe()

Year  Home Team Goals  Away Team Goals     Attendance  \
count   852.000000       852.000000       852.000000     850.000000   
mean   1985.089202         1.811033         1.022300   45164.800000   
std      22.448825         1.610255         1.087573   23485.249247   
min    1930.000000         0.000000         0.000000    2000.000000   
25%    1970.000000         1.000000         0.000000   30000.000000   
50%    1990.000000         2.000000         1.000000   41579.500000   
75%    2002.000000         3.000000         2.000000   61374.500000   
max    2014.000000        10.000000         7.000000  173850.000000   

       Half-time Home Goals  Half-time Away Goals       RoundID       MatchID  
count            852.000000            852.000000  8.520000e+02  8.520000e+02  
mean               0.708920              0.428404  1.066177e+07  6.134687e+07  
std                0.937414              0.691252  2.729613e+07  1.110572e+08  
min                0.000000              0.000000  2.010000e+02  2.500000e+01  
25%                0.000000              0.000000  2.620000e+02  1.188750e+03  
50%                0.000000              0.000000  3.370000e+02  2.191000e+03  
75%                1.000000              1.000000  2.497220e+05  4.395006e+07  
max                6.000000              5.000000  9.741060e+07  3.001865e+08

These describe function will give the infromation about all the statistics 

In [129]:
df1.isnull().sum()

Year                    3720
Datetime                3720
Stage                   3720
Stadium                 3720
City                    3720
Home Team Name          3720
Home Team Goals         3720
Away Team Goals         3720
Away Team Name          3720
Win conditions          3720
Attendance              3722
Half-time Home Goals    3720
Half-time Away Goals    3720
Referee                 3720
Assistant 1             3720
Assistant 2             3720
RoundID                 3720
MatchID                 3720
Home Team Initials      3720
Away Team Initials      3720
dtype: int64

From the above observation I can see that there are 3720 null value.

In [130]:
df1.dropna(inplace=True)

Here i am droping all the null values.


In [131]:
df1.shape

(850, 20)

After droping I am checking again the shape of my data.

In [132]:
# checking all the unique element in my data
df1.head()

Year              Datetime    Stage         Stadium         City  \
0  1930.0  13 Jul 1930 - 15:00   Group 1         Pocitos  Montevideo    
1  1930.0  13 Jul 1930 - 15:00   Group 4  Parque Central  Montevideo    
2  1930.0  14 Jul 1930 - 12:45   Group 2  Parque Central  Montevideo    
3  1930.0  14 Jul 1930 - 14:50   Group 3         Pocitos  Montevideo    
4  1930.0  15 Jul 1930 - 16:00   Group 1  Parque Central  Montevideo    

  Home Team Name  Home Team Goals  Away Team Goals Away Team Name  \
0         France              4.0              1.0         Mexico   
1            USA              3.0              0.0        Belgium   
2     Yugoslavia              2.0              1.0         Brazil   
3        Romania              3.0              1.0           Peru   
4      Argentina              1.0              0.0         France   

  Win conditions  Attendance  Half-time Home Goals  Half-time Away Goals  \
0                     4444.0                   3.0                   0.0   
1                    18346.0                   2.0                   0.0   
2                    24059.0                   2.0                   0.0   
3                     2549.0                   1.0                   0.0   
4                    23409.0                   0.0                   0.0   

                  Referee               Assistant 1  \
0  LOMBARDI Domingo (URU)     CRISTOPHE Henry (BEL)   
1       MACIAS Jose (ARG)  MATEUCCI Francisco (URU)   
2     TEJADA Anibal (URU)   VALLARINO Ricardo (URU)   
3   WARNKEN Alberto (CHI)       LANGENUS Jean (BEL)   
4     REGO Gilberto (BRA)      SAUCEDO Ulises (BOL)   

                  Assistant 2  RoundID  MatchID Home Team Initials  \
0         REGO Gilberto (BRA)    201.0   1096.0                FRA   
1       WARNKEN Alberto (CHI)    201.0   1090.0                USA   
2         BALWAY Thomas (FRA)    201.0   1093.0                YUG   
3    MATEUCCI Francisco (URU)    201.0   1098.0                ROU   
4  RADULESCU Constantin (ROU)    201.0   1085.0                ARG   

  Away Team Initials  
0                MEX  
1                BEL  
2                BRA  
3                PER  
4                FRA

In [133]:

df1['Stage'].unique()

array(['Group 1', 'Group 4', 'Group 2', 'Group 3', 'Semi-finals', 'Final',
       'Preliminary round', 'Quarter-finals', 'Match for third place',
       'First round', 'Group 6', 'Group B', 'Group A', 'Group 5',
       'Group C', 'Group D', 'Group F', 'Group E', 'Round of 16',
       'Group H', 'Group G', 'Third place', 'Play-off for third place'],
      dtype=object)

From the above In stage column there are 23 different unique values.

In [134]:
df1['Stadium'].unique()

array(['Pocitos', 'Parque Central', 'Estadio Centenario',
       'Stadio Benito Mussolini', 'Giorgio Ascarelli', 'San Siro',
       'Littorale', 'Giovanni Berta', 'Luigi Ferraris', 'Nazionale PNF',
       'Littorio', 'Parc des Princes', 'Velodrome Municipale',
       'Stade Olympique', 'Stade Municipal', 'Stade V�lodrome',
       'Stade de la Meinau', 'Cavee Verte', 'Stade du Parc Lescure',
       'Victor Boucquey', 'Fort Carree',
       'Maracan� - Est�dio Jornalista M�rio Filho', 'Durival de Brito',
       'Pacaembu', 'Independencia', 'Eucaliptos', 'Ilha do Retiro',
       'Wankdorf Stadium', 'Hardturm', 'Charmilles', 'La Pontaise',
       'St. Jakob', 'Comunale di Cornaredo', 'Rasunda Stadium',
       'Nya Ullevi', 'Malmo Stadion', 'Jarnvallen', 'Idrottsparken',
       'Arosvallen', 'Rimnersvallen', 'Orjans Vall', 'Ryavallen',
       'Olympia Stadium', 'Tunavallen', 'Eyravallen', 'Carlos Dittborn',
       'Estadio Sausalito', 'Estadio El Teniente-Codelco', 'Nacional',
       'Wemble

In [135]:
df1['Stadium'] = df1['Stadium']. replace(['Stade V�lodrome','Maracan� - Est�dio Jornalista M�rio Filho',
                                          'Nou Camp - Estadio Le�n','Estadio Jos� Mar�a Minella'],
                                   ['Stade Vlodrome','Estdio Jornalista Mrio Filho','Nou Camp - Estadio Len',
                                    'Estadio Jos Mara Minella'])

In [136]:
df1['Stadium'].unique()

array(['Pocitos', 'Parque Central', 'Estadio Centenario',
       'Stadio Benito Mussolini', 'Giorgio Ascarelli', 'San Siro',
       'Littorale', 'Giovanni Berta', 'Luigi Ferraris', 'Nazionale PNF',
       'Littorio', 'Parc des Princes', 'Velodrome Municipale',
       'Stade Olympique', 'Stade Municipal', 'Stade Vlodrome',
       'Stade de la Meinau', 'Cavee Verte', 'Stade du Parc Lescure',
       'Victor Boucquey', 'Fort Carree', 'Estdio Jornalista Mrio Filho',
       'Durival de Brito', 'Pacaembu', 'Independencia', 'Eucaliptos',
       'Ilha do Retiro', 'Wankdorf Stadium', 'Hardturm', 'Charmilles',
       'La Pontaise', 'St. Jakob', 'Comunale di Cornaredo',
       'Rasunda Stadium', 'Nya Ullevi', 'Malmo Stadion', 'Jarnvallen',
       'Idrottsparken', 'Arosvallen', 'Rimnersvallen', 'Orjans Vall',
       'Ryavallen', 'Olympia Stadium', 'Tunavallen', 'Eyravallen',
       'Carlos Dittborn', 'Estadio Sausalito',
       'Estadio El Teniente-Codelco', 'Nacional', 'Wembley Stadium',
       'H

In stadium column there are total 181 unique stadium name.

In [137]:
df1['City'].unique()

array(['Montevideo ', 'Turin ', 'Naples ', 'Milan ', 'Bologna ',
       'Florence ', 'Genoa ', 'Rome ', 'Trieste ', 'Paris ', 'Reims ',
       'Colombes ', 'Toulouse ', 'Marseilles ', 'Strasbourg ',
       'Le Havre ', 'Bordeaux ', 'Lille ', 'Antibes ', 'Rio De Janeiro ',
       'Curitiba ', 'Sao Paulo ', 'Belo Horizonte ', 'Porto Alegre ',
       'Recife ', 'Berne ', 'Zurich ', 'Geneva ', 'Lausanne ', 'Basel ',
       'Lugano ', 'Solna ', 'Gothenburg ', 'Malm� ', 'Sandviken ',
       'Norrk�Ping ', 'Vasteras ', 'Udevalla ', 'Halmstad ', 'Boras ',
       'Helsingborg ', 'Eskilstuna ', 'Orebro ', 'Arica ',
       'Vina Del Mar ', 'Rancagua ', 'Santiago De Chile ', 'London ',
       'Sheffield ', 'Liverpool ', 'Middlesbrough ', 'Manchester ',
       'Birmingham ', 'Sunderland ', 'Mexico City ', 'Puebla ', 'Leon ',
       'Guadalajara ', 'Toluca ', 'Frankfurt/Main ', 'Berlin West ',
       'Hamburg ', 'Dortmund ', 'Hanover ', 'D�Sseldorf ', 'Munich ',
       'Stuttgart ', 'Gelsenkirchen '

From the above we can see that there are in total 151 unique cities.

In [138]:
df1['City'] = df1['City']. replace(['Norrk�Ping ','La Coru�A ','D�Sseldorf '],
                                   ['Norrk Ping','La Coru A', 'D Sseldrof'])

In [139]:
df1['City'].unique()

array(['Montevideo ', 'Turin ', 'Naples ', 'Milan ', 'Bologna ',
       'Florence ', 'Genoa ', 'Rome ', 'Trieste ', 'Paris ', 'Reims ',
       'Colombes ', 'Toulouse ', 'Marseilles ', 'Strasbourg ',
       'Le Havre ', 'Bordeaux ', 'Lille ', 'Antibes ', 'Rio De Janeiro ',
       'Curitiba ', 'Sao Paulo ', 'Belo Horizonte ', 'Porto Alegre ',
       'Recife ', 'Berne ', 'Zurich ', 'Geneva ', 'Lausanne ', 'Basel ',
       'Lugano ', 'Solna ', 'Gothenburg ', 'Malm� ', 'Sandviken ',
       'Norrk Ping', 'Vasteras ', 'Udevalla ', 'Halmstad ', 'Boras ',
       'Helsingborg ', 'Eskilstuna ', 'Orebro ', 'Arica ',
       'Vina Del Mar ', 'Rancagua ', 'Santiago De Chile ', 'London ',
       'Sheffield ', 'Liverpool ', 'Middlesbrough ', 'Manchester ',
       'Birmingham ', 'Sunderland ', 'Mexico City ', 'Puebla ', 'Leon ',
       'Guadalajara ', 'Toluca ', 'Frankfurt/Main ', 'Berlin West ',
       'Hamburg ', 'Dortmund ', 'Hanover ', 'D Sseldrof', 'Munich ',
       'Stuttgart ', 'Gelsenkirchen ', 

In [140]:
df1['Home Team Name'].unique()

array(['France', 'USA', 'Yugoslavia', 'Romania', 'Argentina', 'Chile',
       'Uruguay', 'Brazil', 'Paraguay', 'Austria', 'Hungary',
       'Switzerland', 'Sweden', 'Germany', 'Spain', 'Italy',
       'Czechoslovakia', 'Cuba', 'England', 'Germany FR', 'Turkey',
       'Soviet Union', 'Northern Ireland', 'Mexico', 'Wales', 'Portugal',
       'Korea DPR', 'Peru', 'Belgium', 'Bulgaria', 'German DR', 'Zaire',
       'Poland', 'Australia', 'Scotland', 'Netherlands', 'Haiti',
       'Tunisia', 'Algeria', 'Honduras', 'Canada', 'Morocco',
       'Korea Republic', 'Iraq', 'Denmark', 'rn">United Arab Emirates',
       'Costa Rica', 'Cameroon', 'rn">Republic of Ireland', 'Colombia',
       'Norway', 'Nigeria', 'Saudi Arabia', 'Bolivia', 'Russia', 'Greece',
       'Jamaica', 'South Africa', 'Japan', 'Croatia', 'China PR',
       'Senegal', 'Slovenia', 'Ecuador', 'rn">Trinidad and Tobago',
       'rn">Serbia and Montenegro', 'Angola', 'Czech Republic', 'Togo',
       'Iran', "C�te d'Ivoire", 'Ghana

In [141]:
df1['Home Team Name'] = df1['Home Team Name']. replace([ "C�te d'Ivoire"],
                                   ["C te d' Ivoire"])

In [142]:
df1['Home Team Name'].unique()

array(['France', 'USA', 'Yugoslavia', 'Romania', 'Argentina', 'Chile',
       'Uruguay', 'Brazil', 'Paraguay', 'Austria', 'Hungary',
       'Switzerland', 'Sweden', 'Germany', 'Spain', 'Italy',
       'Czechoslovakia', 'Cuba', 'England', 'Germany FR', 'Turkey',
       'Soviet Union', 'Northern Ireland', 'Mexico', 'Wales', 'Portugal',
       'Korea DPR', 'Peru', 'Belgium', 'Bulgaria', 'German DR', 'Zaire',
       'Poland', 'Australia', 'Scotland', 'Netherlands', 'Haiti',
       'Tunisia', 'Algeria', 'Honduras', 'Canada', 'Morocco',
       'Korea Republic', 'Iraq', 'Denmark', 'rn">United Arab Emirates',
       'Costa Rica', 'Cameroon', 'rn">Republic of Ireland', 'Colombia',
       'Norway', 'Nigeria', 'Saudi Arabia', 'Bolivia', 'Russia', 'Greece',
       'Jamaica', 'South Africa', 'Japan', 'Croatia', 'China PR',
       'Senegal', 'Slovenia', 'Ecuador', 'rn">Trinidad and Tobago',
       'rn">Serbia and Montenegro', 'Angola', 'Czech Republic', 'Togo',
       'Iran', "C te d' Ivoire", 'Ghan

There are in total 78 different home teams

In [143]:
df1['Away Team Name'].unique()

array(['Mexico', 'Belgium', 'Brazil', 'Peru', 'France', 'Bolivia',
       'Paraguay', 'Romania', 'Chile', 'USA', 'Yugoslavia', 'Argentina',
       'Egypt', 'Netherlands', 'Switzerland', 'Sweden', 'Spain',
       'Hungary', 'Austria', 'Germany', 'Czechoslovakia',
       'Dutch East Indies', 'Norway', 'Poland', 'Cuba', 'Italy',
       'England', 'Scotland', 'Turkey', 'Korea Republic', 'Germany FR',
       'Uruguay', 'Wales', 'Northern Ireland', 'Soviet Union', 'Colombia',
       'Bulgaria', 'Korea DPR', 'Portugal', 'Israel', 'Morocco',
       'El Salvador', 'Australia', 'Haiti', 'German DR', 'Zaire',
       'IR Iran', 'Tunisia', 'Iran', 'Cameroon', 'New Zealand', 'Algeria',
       'Honduras', 'Kuwait', 'Iraq', 'Denmark', 'Canada',
       'rn">Republic of Ireland', 'rn">United Arab Emirates',
       'Costa Rica', 'Saudi Arabia', 'Russia', 'Greece', 'Nigeria',
       'South Africa', 'Japan', 'Croatia', 'Jamaica', 'Senegal',
       'Slovenia', 'Ecuador', 'China PR', "C�te d'Ivoire",
       

In [144]:
df1['Away Team Name'] = df1['Away Team Name']. replace([ "C�te d'Ivoire"],
                                   ["C te d' Ivoire"])

In [145]:
df1['Away Team Name'].unique()

array(['Mexico', 'Belgium', 'Brazil', 'Peru', 'France', 'Bolivia',
       'Paraguay', 'Romania', 'Chile', 'USA', 'Yugoslavia', 'Argentina',
       'Egypt', 'Netherlands', 'Switzerland', 'Sweden', 'Spain',
       'Hungary', 'Austria', 'Germany', 'Czechoslovakia',
       'Dutch East Indies', 'Norway', 'Poland', 'Cuba', 'Italy',
       'England', 'Scotland', 'Turkey', 'Korea Republic', 'Germany FR',
       'Uruguay', 'Wales', 'Northern Ireland', 'Soviet Union', 'Colombia',
       'Bulgaria', 'Korea DPR', 'Portugal', 'Israel', 'Morocco',
       'El Salvador', 'Australia', 'Haiti', 'German DR', 'Zaire',
       'IR Iran', 'Tunisia', 'Iran', 'Cameroon', 'New Zealand', 'Algeria',
       'Honduras', 'Kuwait', 'Iraq', 'Denmark', 'Canada',
       'rn">Republic of Ireland', 'rn">United Arab Emirates',
       'Costa Rica', 'Saudi Arabia', 'Russia', 'Greece', 'Nigeria',
       'South Africa', 'Japan', 'Croatia', 'Jamaica', 'Senegal',
       'Slovenia', 'Ecuador', 'China PR', "C te d' Ivoire",
      

In total there are 83 away team or other teams

In [146]:
len(df1['Assistant 1'].unique())

387

There are 387 Assistant 1 in total world cup

In [147]:
len(df1['Assistant 2'].unique())

408

There are 408 Assistant 2 in total world cup

In [148]:
len(df1['Referee'].unique())

366

There are 366 Referee in total world cup

In [149]:
df1['Home Team Initials'].unique()

array(['FRA', 'USA', 'YUG', 'ROU', 'ARG', 'CHI', 'URU', 'BRA', 'PAR',
       'AUT', 'HUN', 'SUI', 'SWE', 'GER', 'ESP', 'ITA', 'TCH', 'CUB',
       'ENG', 'FRG', 'TUR', 'URS', 'NIR', 'MEX', 'WAL', 'POR', 'PRK',
       'PER', 'BEL', 'BUL', 'GDR', 'ZAI', 'POL', 'AUS', 'SCO', 'NED',
       'HAI', 'TUN', 'ALG', 'HON', 'CAN', 'MAR', 'KOR', 'IRQ', 'DEN',
       'UAE', 'CRC', 'CMR', 'IRL', 'COL', 'NOR', 'NGA', 'KSA', 'BOL',
       'RUS', 'GRE', 'JAM', 'RSA', 'JPN', 'CRO', 'CHN', 'SEN', 'SVN',
       'ECU', 'TRI', 'SCG', 'ANG', 'CZE', 'TOG', 'IRN', 'CIV', 'GHA',
       'UKR', 'SRB', 'NZL', 'SVK', 'BIH'], dtype=object)

There are in total 77 country with there code in home teams

In [150]:
df1['Away Team Initials'].unique()

array(['MEX', 'BEL', 'BRA', 'PER', 'FRA', 'BOL', 'PAR', 'ROU', 'CHI',
       'USA', 'YUG', 'ARG', 'EGY', 'NED', 'SUI', 'SWE', 'ESP', 'HUN',
       'AUT', 'GER', 'TCH', 'INH', 'NOR', 'POL', 'CUB', 'ITA', 'ENG',
       'SCO', 'TUR', 'KOR', 'FRG', 'URU', 'WAL', 'NIR', 'URS', 'COL',
       'BUL', 'PRK', 'POR', 'ISR', 'MAR', 'SLV', 'AUS', 'HAI', 'GDR',
       'ZAI', 'IRN', 'TUN', 'CMR', 'NZL', 'ALG', 'HON', 'KUW', 'IRQ',
       'DEN', 'CAN', 'IRL', 'UAE', 'CRC', 'KSA', 'RUS', 'GRE', 'NGA',
       'RSA', 'JPN', 'CRO', 'JAM', 'SEN', 'SVN', 'ECU', 'CHN', 'CIV',
       'CZE', 'GHA', 'TOG', 'UKR', 'TRI', 'SCG', 'ANG', 'SVK', 'SRB',
       'BIH'], dtype=object)

There are 82 away teams in match

In [151]:
# Let's print all the county len
print(df1['Home Team Initials'].unique())


['FRA' 'USA' 'YUG' 'ROU' 'ARG' 'CHI' 'URU' 'BRA' 'PAR' 'AUT' 'HUN' 'SUI'
 'SWE' 'GER' 'ESP' 'ITA' 'TCH' 'CUB' 'ENG' 'FRG' 'TUR' 'URS' 'NIR' 'MEX'
 'WAL' 'POR' 'PRK' 'PER' 'BEL' 'BUL' 'GDR' 'ZAI' 'POL' 'AUS' 'SCO' 'NED'
 'HAI' 'TUN' 'ALG' 'HON' 'CAN' 'MAR' 'KOR' 'IRQ' 'DEN' 'UAE' 'CRC' 'CMR'
 'IRL' 'COL' 'NOR' 'NGA' 'KSA' 'BOL' 'RUS' 'GRE' 'JAM' 'RSA' 'JPN' 'CRO'
 'CHN' 'SEN' 'SVN' 'ECU' 'TRI' 'SCG' 'ANG' 'CZE' 'TOG' 'IRN' 'CIV' 'GHA'
 'UKR' 'SRB' 'NZL' 'SVK' 'BIH']


In [152]:
df1.head()

Year              Datetime    Stage         Stadium         City  \
0  1930.0  13 Jul 1930 - 15:00   Group 1         Pocitos  Montevideo    
1  1930.0  13 Jul 1930 - 15:00   Group 4  Parque Central  Montevideo    
2  1930.0  14 Jul 1930 - 12:45   Group 2  Parque Central  Montevideo    
3  1930.0  14 Jul 1930 - 14:50   Group 3         Pocitos  Montevideo    
4  1930.0  15 Jul 1930 - 16:00   Group 1  Parque Central  Montevideo    

  Home Team Name  Home Team Goals  Away Team Goals Away Team Name  \
0         France              4.0              1.0         Mexico   
1            USA              3.0              0.0        Belgium   
2     Yugoslavia              2.0              1.0         Brazil   
3        Romania              3.0              1.0           Peru   
4      Argentina              1.0              0.0         France   

  Win conditions  Attendance  Half-time Home Goals  Half-time Away Goals  \
0                     4444.0                   3.0                   0.0   
1                    18346.0                   2.0                   0.0   
2                    24059.0                   2.0                   0.0   
3                     2549.0                   1.0                   0.0   
4                    23409.0                   0.0                   0.0   

                  Referee               Assistant 1  \
0  LOMBARDI Domingo (URU)     CRISTOPHE Henry (BEL)   
1       MACIAS Jose (ARG)  MATEUCCI Francisco (URU)   
2     TEJADA Anibal (URU)   VALLARINO Ricardo (URU)   
3   WARNKEN Alberto (CHI)       LANGENUS Jean (BEL)   
4     REGO Gilberto (BRA)      SAUCEDO Ulises (BOL)   

                  Assistant 2  RoundID  MatchID Home Team Initials  \
0         REGO Gilberto (BRA)    201.0   1096.0                FRA   
1       WARNKEN Alberto (CHI)    201.0   1090.0                USA   
2         BALWAY Thomas (FRA)    201.0   1093.0                YUG   
3    MATEUCCI Francisco (URU)    201.0   1098.0                ROU   
4  RADULESCU Constantin (ROU)    201.0   1085.0                ARG   

  Away Team Initials  
0                MEX  
1                BEL  
2                BRA  
3                PER  
4                FRA

In [153]:
df1['Home Team Name'].unique()

array(['France', 'USA', 'Yugoslavia', 'Romania', 'Argentina', 'Chile',
       'Uruguay', 'Brazil', 'Paraguay', 'Austria', 'Hungary',
       'Switzerland', 'Sweden', 'Germany', 'Spain', 'Italy',
       'Czechoslovakia', 'Cuba', 'England', 'Germany FR', 'Turkey',
       'Soviet Union', 'Northern Ireland', 'Mexico', 'Wales', 'Portugal',
       'Korea DPR', 'Peru', 'Belgium', 'Bulgaria', 'German DR', 'Zaire',
       'Poland', 'Australia', 'Scotland', 'Netherlands', 'Haiti',
       'Tunisia', 'Algeria', 'Honduras', 'Canada', 'Morocco',
       'Korea Republic', 'Iraq', 'Denmark', 'rn">United Arab Emirates',
       'Costa Rica', 'Cameroon', 'rn">Republic of Ireland', 'Colombia',
       'Norway', 'Nigeria', 'Saudi Arabia', 'Bolivia', 'Russia', 'Greece',
       'Jamaica', 'South Africa', 'Japan', 'Croatia', 'China PR',
       'Senegal', 'Slovenia', 'Ecuador', 'rn">Trinidad and Tobago',
       'rn">Serbia and Montenegro', 'Angola', 'Czech Republic', 'Togo',
       'Iran', "C te d' Ivoire", 'Ghan

In [154]:
df1['Home Team Name'] = df1['Home Team Name']. replace(['''rn">Trinidad and Tobago''',"C te d' Ivoire",
                                                        'rn">Bosnia and Herzegovina'],
                                   ['Trinidad and Tobago','Ctedivore','Bosnia and Herzegovina'])

# The end of stage 1

In [155]:
import matplotlib.pyplot as plt
import seaborn as sns
import plotly as py                
import cufflinks as cf
%matplotlib inline
from plotly.offline import iplot   
py.offline.init_notebook_mode(connected=True)
cf.go_offline()


In [156]:
df3.head()

Year      Country      Winner      Runners-Up    Third      Fourth  \
0  1930      Uruguay     Uruguay       Argentina      USA  Yugoslavia   
1  1934        Italy       Italy  Czechoslovakia  Germany     Austria   
2  1938       France       Italy         Hungary   Brazil      Sweden   
3  1950       Brazil     Uruguay          Brazil   Sweden       Spain   
4  1954  Switzerland  Germany FR         Hungary  Austria     Uruguay   

   GoalsScored  QualifiedTeams  MatchesPlayed Attendance  
0           70              13             18    590.549  
1           70              16             17    363.000  
2           84              15             18    375.700  
3           88              13             22  1.045.246  
4          140              16             26    768.607

Most of the world cup winning

In [157]:
win = df3['Winner'].value_counts()

In [158]:
win.iplot(kind='bar')

In [159]:
goal = df3['GoalsScored'].value_counts()
goal

70     2
171    2
89     2
146    1
147    1
161    1
141    1
115    1
132    1
102    1
84     1
97     1
95     1
126    1
140    1
88     1
145    1
Name: GoalsScored, dtype: int64

In [167]:
import plotly.express as px
fig = px.pie(df3, values=df3['GoalsScored'], names=df3['Winner'])
fig.show()
